rkt/enter.go

```diff
@@ -0,0 +1,174 @@
+		printCommandUsageByName(cmdEnterName)
+		return 1
+	}
+
+	containerUUID, err := types.NewUUID(args[0])
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Invalid UUID: %v\n", err)
+		return 1
+	}
+
+	cid := containerUUID.String()
+	cdir := filepath.Join(containersDir(), cid)
+
+	if err = pingContainer(cdir); err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to query container %q: %v\n", cid, err)
+		return 1
+	}
+
+	imageID, err := getAppImageID(cdir)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to determine image id: %v\n", err)
+		return 1
+	}
+
+	_, err = os.Stat(filepath.Join(rktpath.AppRootfsPath(cdir, *imageID)))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to access app rootfs: %v\n", err)
+		return 1
+	}
+
+	argv, err := getEnterArgv(cdir, imageID, args)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Enter failed: %v\n", err)
+		return 1
+	}
+
+	err = stage0.Enter(cdir, imageID, argv)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Enter failed: %v\n", err)
+		return 1
+	}
+	// not reached when stage0.Enter execs /enter
+	return 0
+}
+
+// getAppImageID returns the image id to enter
+// If one was supplied in the flags then it's simply returned
+// If the CRM contains a single image, that image's id is returned
+// If the CRM has multiple images, the ids and names are printed and an error is returned
+func getAppImageID(cdir string) (*types.Hash, error) {
+	if !flagAppImageID.Empty() {
+		return &flagAppImageID, nil
+	}
+
+	// figure out the image id, or show a list if multiple are present
+	b, err := ioutil.ReadFile(rktpath.ContainerManifestPath(cdir))
+	if err != nil {
+		return nil, fmt.Errorf("error reading container manifest: %v", err)
+	}
+
+	m := schema.ContainerRuntimeManifest{}
+	err = m.UnmarshalJSON(b)
+	if err != nil {
+		return nil, fmt.Errorf("unable to load manifest: %v", err)
+	}
+
+	switch len(m.Apps) {
+	case 0:
+		return nil, fmt.Errorf("container contains zero apps")
+	case 1:
+		return &m.Apps[0].ImageID, nil
+	default:
+	}
+
+	fmt.Fprintf(os.Stderr, "Container contains multiple apps:\n")
+	for _, ra := range m.Apps {
+		fmt.Fprintf(os.Stderr, "\t%s: %s\n", types.ShortHash(ra.ImageID.String()), ra.Name.String())
+	}
+
+	return nil, fmt.Errorf("specify app using \"rkt enter --imageid ...\"")
+}
+
+// pingContainer checks to see if the container is running
+func pingContainer(cdir string) error {
+	l, err := lock.TrySharedLock(cdir)
+	switch err {
+	case nil:
+		l.Close()
+		return fmt.Errorf("inactive")
+	case lock.ErrNotExist:
+		return fmt.Errorf("nonexistent")
+	case lock.ErrPermission:
+		return fmt.Errorf("access denied: %v", err)
+	default:
+		return err
+	case lock.ErrLocked:
+	}
+
+	return nil
+}
+
+// getEnterArgv returns the argv to use for entering the container
+func getEnterArgv(cdir string, imageID *types.Hash, cmdArgs []string) ([]string, error) {
+	argv := []string{}
+	if len(cmdArgs) < 2 {
+		fmt.Printf("No command specified, assuming %q\n", defaultCmd)
+		argv = append(argv, defaultCmd)
```

Comment:
Why append? Just assign: `argv = []string{ defaultCmd }`. Also, change above to just `var argv []string`.